### ISSUE TITLE
Error in qhull_ra.h when building with icc

### ISSUE THREAD

**Issue by wscullin** [qhull/qhull#24]
I am building qhull 2015.2 with Intel C++ 2017.2 (17.0.2.174) on SUSE Linux Enterprise Server 12 and receive the following error:

```
In file included from /stage/qhull-2015.2-build/qhull-2015.2/src/libqhullcpp/Coordinates.h(13),
                 from /stage/qhull-2015.2-build/qhull-2015.2/src/libqhullcpp/Coordinates.cpp(9):
/stage/qhull-2015.2-build/qhull-2015.2/src/libqhull_r/qhull_ra.h(105): error: this declaration may not have extern "C" linkage
  template <typename T>
```

**Comment by alalazo** [comment]
I am currently seeing the same on `rhel6` and `rhel7` with Intel 2017.2.

**Comment by alalazo** [comment]
@qhull This issue is a duplicate of #16 and was apparently solved by #8. Do you plan releasing a new version with the fix?

**Comment by qhull** [comment]
There will not be a new version soon, so a patch 
may be appropriate.  I'm not a regular git user, 
so please select the mechanism that you think best.

I expect the next release of Qhull to be near the 
end of this year or early next year.

                                 --Brad

At 10:21 AM 7/18/2017, Massimiliano Culpo wrote:

**Comment by alalazo** [comment]
@qhull Thanks for the update.

**Comment by tungmilan** [comment]
@cbbarber : is there any new ETA for the new release? Thanks!

**Comment by cbbarber** [comment]
A new release is expected early next year.  A development release of libqhull_r will occur sometime next month.

Implementing vertex merging to fix "Nearly coincident points on an edge" in Limitations of Merged Facets [http://www.qhull.org/html/qh-impre.htm#limit] revealed further issues with facet merging in 4-D and higher.  While these issues are unlikely in practice, they are well-known (see "Post-merging of coplanar facets" and "Maximum facet width"  under Limitations) .  

In 4-D and higher, the convexity test for non-simplicial facets needs to consider the facet's vertices as well as its centrum.   If only the centrum is tested, a "twisted" facet may incorrectly pass the convexity test.  These failures are likely if vertices are frequently nearby but not coincident (e.g., 3-d Delaunay triangulations of 200 nearby triplets, 'rbox 200 C2,1e-13 t | qdelaunay Tv').

                                        --Brad

**Comment by cbbarber** [comment]
Fixed in developer release Qhull 2018.0.1.   Qhull 2019.1 will be released later this month.

 - Move 'extern "C" {}' logic from C++ to C headers [C. Atkins, wscullin, alalazo]
